webtester-core/src/test/java/integration/pageobjects/MultiSelectIntegrationTest.java

```diff
@@ -0,0 +1,290 @@
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public final void testThatCorrectExceptionIsThrownBySelectingWithUnknownText() {
+        page.multiSelect.selectByText("unknown");
+    }
+
+    /* select by value */
+
+    @Test
+    public final void testThatNothingChangesIfNoValuesAreGivenAsParameter() {
+        page.multiSelectWithSelection.selectByValues();
+        assertThat(page.multiSelectWithSelection.getAllSelectedValues(), contains("2", "3"));
+    }
+
+    @Test
+    public final void testThatOptionsAreSelectedCorrectlyByValues_multiSelect() {
+        page.multiSelect.selectByValues("1", "3");
+        assertThat(page.multiSelect.getAllSelectedValues(), contains("1", "3"));
+    }
+
+    @Test
+    public final void testThatSelectionByValuesIsAdditive_multiSelectWithSelection() {
+        page.multiSelectWithSelection.selectByValues("1");
+        assertThat(page.multiSelectWithSelection.getAllSelectedValues(), contains("1", "2", "3"));
+    }
+
+    @Test
+    public final void testThatDoubleSelectByValueDoNotDeselectOption_multiSelectWithSelection() {
+        page.multiSelectWithSelection.selectByValues("1", "3");
+        assertThat(page.multiSelectWithSelection.getAllSelectedValues(), contains("1", "2", "3"));
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public final void testThatCorrectExceptionIsThrownByUnknownValue_select() {
+        page.multiSelect.selectByValues("42");
+    }
+
+    /* select by index */
+
+    @Test
+    public final void testThatNothingChangesIfNoIndicesAreGivenAsParameter() {
+        page.multiSelectWithSelection.selectByIndices();
+        assertThat(page.multiSelectWithSelection.getAllSelectedIndices(), contains(1, 2));
+    }
+
+    @Test
+    public final void testThatOptionsAreSelectedCorrectlyByIndices_multiSelect() {
+        page.multiSelect.selectByIndices(0, 2);
+        assertThat(page.multiSelect.getAllSelectedIndices(), contains(0, 2));
+    }
+
+    @Test
+    public final void testThatSelectionByIndicesIsAdditive_multiSelectWithSelection() {
+        page.multiSelectWithSelection.selectByIndices(0);
+        assertThat(page.multiSelectWithSelection.getAllSelectedIndices(), contains(0, 1,2));
+    }
+
+    @Test
+    public final void testThatDoubleSelectByIndexDoNotDeselectOption_multiSelectWithSelection() {
+        page.multiSelectWithSelection.selectByIndices(0, 2);
+        assertThat(page.multiSelectWithSelection.getAllSelectedIndices(), contains(0, 1, 2));
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public final void testThatCorrectExceptionIsThrownByUnknownIndex_select() {
+        page.multiSelect.selectByIndices(42);
+    }
+
+    /* deselect all */
+
+    @Test
+    public final void testThatAllOptionsAreDeselected() {
+        page.multiSelectWithSelection.deselectAll();
+        assertThat(page.multiSelectWithSelection.getAllSelectedOptions().isEmpty(), is(true));
```

Comment:
`assertThat(page.multiSelectWithSelection.getAllSelectedOptions().isEmpty(), is(true));`
>>
`assertThat(page.multiSelectWithSelection.getAllSelectedOptions(), is(empty()));`